### PR TITLE
[Bug 17031] Make sure mobilecomposemail returns a valid 'result'

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

File: engine/src/mbliphonemail.mm [hunk 1]
```diff
@@ -654,5 +654,9 @@ void MCSystemGetCanSendMail(bool& r_result)
 
 void MCSystemMailResult(MCStringRef& r_result)
 {
-	r_result = MCValueRetain(kMCEmptyString);
+	// PM-2016-02-29: [[ Bug 17031 ]] Make sure we return 'the result'
+	if (MCValueGetTypeCode(MCresult -> getvalueref()) == kMCValueTypeCodeString)
+		r_result = MCValueRetain((MCStringRef)MCresult -> getvalueref());
```
Comment (peter-b) [inline engine/src/mbliphonemail.mm]: For the future, sometimes the operator precedence can be clearer by using the C++ constructor-style type cast, e.g.:

``` C++
r_result = MCValueRetain(MCStringRef(MCresult->getvalueref()));
```

**Comment by peter-b** [comment]
@livecode-vulcan review ok c8995e9

**Comment by peter-b** [comment]
@livecode-vulcan review 2ca145a

**Comment by livecodepanos** [comment]
@peter-b I think you missed the `ok` :)

@livecode-vulcan review ok 2ca145a

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok 2ca145ae0146a3bf9d993141fc185a35d37ca880

**Comment by livecode-vulcan** [comment]
😎 test success 2ca145ae0146a3bf9d993141fc185a35d37ca880
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1973)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1986)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1967)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1979)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1989)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1994)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1999)

**Pull request merged**